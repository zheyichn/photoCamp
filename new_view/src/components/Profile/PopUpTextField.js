import TextField from '@mui/material/TextField';
// import { useEffect, useState } from 'react';

export default function PopUpTextField(props) {

    const {field, postInfo, defaultValue} = props;

    let textValue = defaultValue;
    
    if (field === 'imageSrc') {
        postInfo.current.imageSrc = textValue;
    }

    if (field === 'postCaption') {
        postInfo.current.postCaption = textValue;
    }

    const changeRef = (e) => {
        textValue = e.target.value;
        if (field === 'imageSrc') {
            postInfo.current.imageSrc = textValue;
        }
        if (field === 'postCaption') {
            postInfo.current.postCaption = textValue;
        }
    }

    return(
        <TextField fullWidth onChange={changeRef} defaultValue={defaultValue || ''} />
    );
}