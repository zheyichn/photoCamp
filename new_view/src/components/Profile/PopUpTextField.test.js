import { cleanup , render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import renderer from "react-test-renderer";
import PopUpTextField from "./PopUpTextField";
import userEvent from "@testing-library/user-event";

// afterEach function runs after each test suite is executed
afterEach(() => {
  cleanup();
});

describe("PopUpTextField", () => {

  test("check textfield can be changed", () => {
    render(<PopUpTextField />);
    const element = screen.getByRole("textbox");
    userEvent.type(element, "hello");
    expect(element).toHaveValue("hello");
  })

  it("Page renders correctly", () => {
    const tree = renderer.create(<PopUpTextField />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});