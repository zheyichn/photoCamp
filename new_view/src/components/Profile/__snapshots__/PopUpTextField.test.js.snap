// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUpTextField Page renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
      defaultValue=""
      disabled={false}
      id=":r0:"
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="text"
    />
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-ihdtdm"
      >
        <span
          className="notranslate"
        >
          â€‹
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
